chajda/tsquery/augments.py

```diff
@@ -53,114 +58,215 @@ def augments_gensim(lang, word, config=Config(), n=5):
+    Due to the size of fasttext models (>1gb), and the loading time,
+    we do not load all models on program startup.
+    Instead, this function is used to load them lazily as needed.
+    lang: language passed from augments_fasttext
+
+    NOTE:
+    The doctest below has been commented out due to requiring more memory than github actions allows for.
+    # >>> load_fasttext_model('ja') or fasttext_models['ja'] != None
+    # True
+    '''
+
+    # suppress_stdout_stderr() is used for redirecting stdout and stderr when downloading and loading fasttext models.
+    # Without this, stdout and stderr from fasttext causes doctests to fail.
+    with suppress_stdout_stderr():
+
+        # fasttext downloads models to the current directory,
+        # but we want to store the model in chaja/tsquery/fasttext,
+        # so we change the working directory before downloading
+        working_dir = os.getcwd()
+        os.makedirs(f"{fasttext_annoy_data_path}/fasttext", exist_ok=True)
+        os.chdir(f"{fasttext_annoy_data_path}/fasttext")
+
+        # download and load fasttext model
+        fasttext.util.download_model(lang, if_exists='ignore')
+        fasttext_models[lang] = fasttext.load_model(f"cc.{lang}.300.bin")
+
+        # fasttext downloads cc.{lang}.300.bin.gz and then decompresses the .gz file into cc.{lang}.300.bin,
+        # however, we only need cc.{lang}.300.bin,
+        # so we remove unneeded .gz file if exists
+        if os.path.isfile(f"{fasttext_annoy_data_path}/fasttext/cc.{lang}.300.bin.gz"):
+            os.remove(f"{fasttext_annoy_data_path}/fasttext/cc.{lang}.300.bin.gz")
+
+        # change working directory back
+        os.chdir(working_dir)
+
+    # Executing fasttext's get_nearest_neighbor as well as accessing the words in the fasttext model both run
+    # significantly faster after they have been called once,
+    # so we call them once upon loading the model to ensure most efficient runtime for subsequent uses.
+    fasttext_models[lang].get_nearest_neighbors('google', k=5)
+    fasttext_list = []
+    for word in fasttext_models[lang].get_words(on_unicode_error='replace'):
+        fasttext_list.append(word)
+
+
+def destroy_fasttext(lang):
+    '''
+    removes fasttext model from disk and memory
+    '''
+
+    # remove fasttext download from disk
+    if os.path.isfile(f"{fasttext_annoy_data_path}/fasttext/cc.{lang}.300.bin"):
+        os.remove(f"{fasttext_annoy_data_path}/fasttext/cc.{lang}.300.bin")
+
+    # remove fasttext model from memory
+    try:
+        del fasttext_models[lang]
+    except KeyError:
+        pass
+
+
+################################################################################
+# annoy
+################################################################################
+
+# dictionary used for storing language abbreviations as keys,
+# and a namedtuple containing the corresponding annoy index, as well as two json files, as values
 annoy_indices = {}
```

Comment:
the comment above is still pretty verbose; you're also duplicating the information about what's in the named tuple into a comment, which is likely to get out of date; better is
```
# stores all information needed for accessing the annoy index of a particular language
annoy_info = namedtuple('annoy_info', ['index','index_for_word','word_at_index'])

# dictionary with keys of language and values of annoy_info
annoy_indices = {}
```
then you would also modify the code below (starting at line 174 ) so that it looks lilke
```
index = ...
index_for_word = ...
word_at_index = ...
annoy_indices[lang] = annoy_info(index, index_for_word, word_at_index)
```
actually, the way you were using named tuples before was incorrect as well; you need to declare the named tuple once as a new type, then reuse that type like I'm doing above